### PR TITLE
dbhydra data mapping

### PR DESCRIPTION
**Pull request by tobyy22**
implementation as discussed with Dominik, need to discuss Union and Optional mapping

### CODE REVIEW

File: dbhydra/dbhydra_core.py [hunk 1]
```diff
@@ -329,6 +350,12 @@ def close_connection(self):
     def initialize_migrator(self, other_db=None):
         self.migrator = Migrator(self, other_db)
 
+    def _convert_column_type_dict_from_python(self, column_type_dict):
+        typing_python_mapping = {k: self.typing_to_python_mapping.get(v, v) for k, v in column_type_dict.items()}
+        return {k: self.python_database_type_mapping[v] for k, v in typing_python_mapping.items()}
```
Comment (YoungMasterGandalf) [inline dbhydra/dbhydra_core.py:355]: Hard to read and grasp what the method really does. A docstring with an example would be beneficial here I think.

File: dbhydra/dbhydra_core.py [hunk 2]
```diff
@@ -414,6 +445,18 @@ def get_foreign_keys_columns(self):
 
 
 class Mysqldb(AbstractDB):
+
+    python_database_type_mapping = PYTHON_TO_MYSQL_DATA_MAPPING = \
+    {
+    'int': "int",
+    'float': "double",
+    'str': "varchar(255)",
```
Comment (tobyy22) [inline dbhydra/dbhydra_core.py:453]: nvarchar

File: dbhydra/dbhydra_core.py [hunk 3]
```diff
@@ -236,6 +236,27 @@ def extract_columns_and_types_from_df(self, dataframe):
 
 
 class AbstractDB(abc.ABC):
+    typing_to_python_mapping = {
+        'List': 'list',
+        'Dict': 'dict',
+        'Tuple': 'tuple',
+        'Set': 'set',
+        'Union': 'object',
```
Comment (tobyy22) [inline dbhydra/dbhydra_core.py:244]: Union ignore, Optional to be solved with nullable dbhydra

File: dbhydra/dbhydra_core.py [hunk 4]
```diff
@@ -1662,4 +1712,18 @@ def df_to_dict(df, column1, column2):
...
\ No newline at end of file
+    return (df)
+
+from pydantic import BaseModel
+from typing import List, Dict
+
+class AbstractModel(abc.ABC, BaseModel):
+    pass
+
+class DbTableModel(AbstractModel):
```
Comment (DovaX) [inline dbhydra/dbhydra_core.py:1723]: shouldnt be here but in example

**Review by DovaX — APPROVED** [review]

**Pull request merged**